src/VisualStudio/Core/Def/Implementation/Venus/SimpleContainedDocument.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Experiment;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Venus
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+    /// <summary>
+    /// An IVisualStudioDocument which represents the secondary buffer to the workspace API.
+    /// </summary>
+    internal sealed class SimpleContainedDocument : ForegroundThreadAffinitizedObject, IVisualStudioHostDocument
+    {
+        /// <summary>
+        /// The IDocumentProvider that created us.
+        /// </summary>
+        private readonly DocumentProvider _documentProvider;
+        private readonly SourceTextContainer _sourceTextContainer;
+        private readonly EventHandler _updatedHandler;
+
+        public DocumentId Id { get; }
+        public IReadOnlyList<string> Folders { get; }
+        public AbstractProject Project { get; }
+        public SourceCodeKind SourceCodeKind { get; }
+        public DocumentKey Key { get; }
+        public TextLoader Loader { get; }
+        public IDocumentServiceFactory DocumentServiceFactory { get; }
+
+        public SimpleContainedDocument(
+            DocumentProvider documentProvider,
+            AbstractProject project,
+            DocumentKey documentKey,
+            Func<uint, IReadOnlyList<string>> getFolderNames,
+            SourceTextContainer sourceTextContainer,
+            SourceCodeKind sourceCodeKind,
+            DocumentId id,
+            EventHandler updatedHandler,
+            IDocumentServiceFactory documentServiceFactory)
+        {
+            Contract.ThrowIfNull(documentProvider);
+
+            this.Project = project;
+            this.Id = id ?? DocumentId.CreateNewId(project.Id, documentKey.Moniker);
+
+            var itemid = this.GetItemId();
```

Comment:
Another issue here... `GetItemId` expects `this.Key` to be set already (line 64).